core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -399,4 +421,132 @@ public void close() throws IOException {
             IOUtils.close(analyzerProviderFactories.values().stream().map((a) -> ((PreBuiltAnalyzerProviderFactory)a).analyzer()).collect(Collectors.toList()));
         }
     }
+
+    public IndexAnalyzers build(IndexSettings indexSettings,
```

Comment:
Note for anyone else reviewing: these next two methods look like they are just copied from the (now removed) `AnalysisService`.